src/StaticAssets/src/StaticAssetsEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,87 @@
+{
+    /// <summary>
+    /// Maps static files produced during the build as endpoints.
+    /// </summary>
+    /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/>.</param>
+    /// <param name="staticAssetsManifestPath">The path to the static assets manifest file.</param>
+    /// <remarks>
+    /// The <paramref name="staticAssetsManifestPath"/> can be null to use the <see cref="IHostEnvironment.ApplicationName"/> to locate the manifes.
+    /// As an alternative, a full path can be specified to the manifest file. If a relative path is used, we'll search for the file in the <see cref="AppContext.BaseDirectory"/>." />
+    /// </remarks>
+    public static StaticAssetsEndpointConventionBuilder MapStaticAssets(this IEndpointRouteBuilder endpoints, string? staticAssetsManifestPath = null)
+    {
+        ArgumentNullException.ThrowIfNull(endpoints);
+
+        var environment = endpoints.ServiceProvider.GetRequiredService<IWebHostEnvironment>();
+        staticAssetsManifestPath ??= $"{environment.ApplicationName}.staticwebassets.endpoints.json";
+
+        staticAssetsManifestPath = !Path.IsPathRooted(staticAssetsManifestPath) ?
+            Path.Combine(AppContext.BaseDirectory, staticAssetsManifestPath) :
+            staticAssetsManifestPath;
+
+        var result = MapStaticAssetsCore(endpoints, staticAssetsManifestPath);
+
+        if (StaticAssetDevelopmentRuntimeHandler.IsEnabled(endpoints.ServiceProvider, environment))
+        {
+            StaticAssetDevelopmentRuntimeHandler.EnableSupport(endpoints, result, environment, result.Descriptors);
+        }
+
+        return result;
+    }
+
+    private static StaticAssetsEndpointConventionBuilder MapStaticAssetsCore(
+        IEndpointRouteBuilder endpoints,
+        string manifestPath,
+        StaticAssetsManifest? manifest = null)
+    {
+        foreach (var ds in endpoints.DataSources)
+        {
+            if (ds is StaticAssetsEndpointDataSource sads && sads.ManifestPath.Equals(manifestPath, StringComparison.Ordinal))
+            {
+                return sads.DefaultBuilder;
+            }
+        }
+
+        if (manifest == null && !File.Exists(manifestPath))
+        {
+            throw new InvalidOperationException($"The static resources manifest file '{manifestPath}' was not found.");
+        }
+
+        manifest ??= StaticAssetsManifest.Parse(manifestPath);
+
+        var dataSource = manifest.CreateDataSource(endpoints, manifestPath, manifest.Endpoints);
+        return dataSource.DefaultBuilder;
+    }
+
+    // For testing purposes
+    internal static StaticAssetsEndpointConventionBuilder MapStaticAssets(this IEndpointRouteBuilder endpoints, StaticAssetsManifest manifest)
+    {
+        ArgumentNullException.ThrowIfNull(endpoints);
+
+        var environment = endpoints.ServiceProvider.GetRequiredService<IWebHostEnvironment>();
+        var result = MapStaticAssetsCore(endpoints, "unused", manifest);
```

Comment:
So `MapStaticAssetsCore` only uses its 2nd _or_ 3rd argument, but never both (ignoring the 2nd argument if the 3rd is specified)? I wonder if it would make sense to split that method into two different ones that accept only the arguments they need. Not hugely important but threw me off a bit when reviewing.

Reply:
I see what you mean, the manifest literal is only used for testing, and I wanted to avoid duplicating the logic. I'm going to leave it as it is as we can clean it up later if we find its problematic.